### ISSUE TITLE
PCIe ASPM conflict does not work with the solution described in the README but it does with other kernel parameter

### ISSUE THREAD

**Issue by gregory112** [tomaspinho/rtl8821ce#53]
Asus ROG Zephyrus GA502DU turns to have RTL8821ce chip. At first the interface does not even show in `ip a` command and it shows up now after the driver is installed. However, the connection to any wifi access point cannot be made, 5GHz or 2.4GHz. `dmesg` reports many power related errors, and `wpa_supplicant` said it fails to turn up the interface. This seems to be related to the PCIe ASPM issue.

This laptop seems to have ASPM active. And the readme clearly states that it can conflict with the driver. Many people seem to have this issue. However, setting this kernel parameter means
```
pci=noaer
```
disabling PCI advanced error reporting, as stated in [https://www.kernel.org/doc/Documentation/admin-guide/kernel-parameters.txt](https://www.kernel.org/doc/Documentation/admin-guide/kernel-parameters.txt)
```
noaer		  [PCIE] If the PCIEAER kernel config parameter is
				enabled, this kernel boot option can be used to
				disable the use of PCIE advanced error reporting.
```
to disable ASPM globally instead, there is this kernel parameter which can be set to forcefully turn on or off ASPM.
```
pcie_aspm=	[PCIE] Forcibly enable or disable PCIe Active State Power
			Management.
		off	Disable ASPM.
		force	Enable ASPM even on devices that claim not to support it.
			WARNING: Forcing ASPM on may cause system lockups.
```
Setting `pcie_aspm=off` kernel parameter seems to fix this power issue. Power warnings still show in `dmesg` but wifi connection can be made now. AFAIK, this however disables system-wide ASPM and may cause the laptop to drain power faster. This documentation seem to provide another detail to disable ASPM per-device.

[https://wireless.wiki.kernel.org/en/users/documentation/aspm#how_to_find_the_link_control_register_on_a_pcie_device](https://wireless.wiki.kernel.org/en/users/documentation/aspm#how_to_find_the_link_control_register_on_a_pcie_device)

Might want to update the README with this solution.

**Comment by tomaspinho** [comment]
Hi @gregory112! I'm in the process of merging much newer sources for this driver at https://github.com/tomaspinho/rtl8821ce/pull/54. Would you mind trying those and see if it fixed the power management problem?

**Comment by gregory112** [comment]
> Hi @gregory112! I'm in the process of merging much newer sources for this driver at #54. Would you mind trying those and see if it fixed the power management problem?

Have tried this in Arch Linux with kernel 5.2.4 and it works. The 5 GHz also works. Also, I tried to use the latest from AUR without the `pcie_aspm` flag and somehow it works fine now. I tried to use this driver with ther latest version that you just merged without PCI advanced error reporting and let PCIe ASPM enabled, and it works.

Guess this issue can be closed now.
